src/Symfony/Component/Config/Resource/FileResource.php

```diff
@@ -29,10 +29,16 @@ class FileResource implements SelfCheckingResourceInterface, \Serializable
      * Constructor.
      *
      * @param string $resource The file path to the resource
+     *
+     * @throws \InvalidArgumentException
      */
     public function __construct($resource)
     {
         $this->resource = realpath($resource);
+
+        if (false === $this->resource) {
+            throw new \InvalidArgumentException(sprintf('The file "%s" does not exist.', $resource));
```

Comment:
The question is how we fix this. I tend to fix the creation of `FileResource` instances to only build these objects when we handle "real" files. However, if we experience these issues in our test suite, isn't this something that will likely appear to other users too and (unexpectedly) break their applications?